python/setup.py

```diff
@@ -235,6 +235,14 @@ def get_packages(self):
         "adag": [
             "cupy-cuda12x; sys_platform != 'darwin'",
         ],
+        "azure": [
+            "azure-cli-core",
+            "azure-identity",
+            "azure-mgmt-compute",
+            "azure-mgmt-network",
+            "azure-mgmt-resource",
+            "msrestazure",
+        ],
```

Comment:
any test?

Reply:
Tested manually.

```sh
conda create -n ray-48847 python=3.9
conda activate ray-48847
cd python/
pip install .
pip list | grep azure # no azure output
pip install .[azure]
pip list | grep azure # has azure output
```

Reply:
then do we need to keep it working for future releases?

Reply:
What do you mean?

Reply:
like, for example, what if some of the package runs into unresolvable dependency conflict with ray's other dependencies. or when `pip install ray[azure]` some other part of ray code starts breaking (like due to `try: import x`)?

Reply:
what I mean is that maybe should add a test in CI in some form?

Reply:
I think it would be fine. We also do not have tests for other extra packages, such as `ray[observability]`, which contains `opentelemetry`.

Reply:
I think you bring up a good point.. `azure` should not be included in `all`

Reply:
we removed azure in our images (for good reasons), now they are being put back in this PR.

Reply:
Good point. I've excluded `ray[azure]` in `ray[all]` in https://github.com/ray-project/ray/pull/48847/commits/2d8ea0d0fd26bda4a0a17068aef75a9f17a637e2.

Reply:
> excluded ray[azure] in ray[all]

> We also do not have tests for other extra packages, such as ray[observability], which contains opentelemetry.

now.. `ray[observability]` is tested via `ray[all]`, and `ray[azure]` is not..


could you provide the context of your goal here?

Reply:
@jjyao told me that I can group Azure-related packages as an extra package. Not sure if he still thinks this way now.